agent/php_execute.c

```diff
@@ -2227,12 +1966,11 @@ static void nr_php_instrument_func_end(NR_EXECUTE_PROTO) {
    */
   if (nrunlikely(OP_ARRAY_IS_A_FILE(NR_OP_ARRAY))) {
     nr_php_execute_file(NR_OP_ARRAY, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
-    php_observer_clear_uncaught_exception_globals();
     return;
   }
 
   /*
-   * Get the current segment and return if null.  The segment would only have
+   * Get the current segment and return if null. The segment would only have
    * been created if we are recording and if wraprec is set or if tt is greater
    * than 0.
```

Comment:
This comment is no longer true, right? When Observer API is used to hook into Zend Engine, the segment is always created in `nr_php_instrument_func_begin` [here](https://github.com/newrelic/newrelic-php-agent/blob/cbeb5ee3eea483d9ccdf7bb6dc0d7cd62f5d0f3d/agent/php_execute.c#L1903).